### DISCUSSION TITLE
How to increase timeout in ClickHouse connector

### DISCUSSION THREAD

**Discussion by JonathanAquino-NextRoll** [trinodb/trino#13531] [Q&A] [answered]
Is there a way to increase the timeout in the ClickHouse connector? I am getting this error after 30 seconds:

```
Query 20220806_164547_02303_ddk5v failed: Read timed out, server ClickHouseNode(addr=http:clickhouse-production.internal.example.com:8123, db=default)@151730631
```

Based on https://github.com/ClickHouse/ClickHouse/issues/974 I think the JDBC setting to change is socket_timeout. It would be great if I could increase this to 300000.

**Comment by tangjiangling** [comment, marked as answer]
You need to change the catalog property `connection-url`(under `etc/catalog/clickhouse.properties`), sth like this:

```
connection-url=jdbc:clickhouse://url:port/default?socket_timeout=300000
```

**Reply by JonathanAquino-NextRoll** [reply]
OK thank you.
